packages/studio-base/src/players/OrderedStampPlayer.ts

```diff
@@ -54,6 +54,24 @@ const getTopicsWithHeader = memoizeWeak((topics: Topic[], datatypes: RosDatatype
   });
 });
 
+function compareMessageEventsByTime(
+  a: MessageEvent<StampedMessage>,
+  b: MessageEvent<StampedMessage>,
+): number {
+  const timeA = getTimestampForMessage(a.message);
+  const timeB = getTimestampForMessage(b.message);
+  if (timeA == undefined && timeB == undefined) {
+    return 0;
+  }
+  if (timeA == undefined) {
+    return -1;
+  }
+  if (timeB == undefined) {
+    return 1;
+  }
+  return compare(timeA, timeB);
+}
```

Comment:
If we assume having timestamps for both messages is the common case, here's a variant that uses a single branch for that case:

```suggestion
function compareMessageEventsByTime(
  a: MessageEvent<StampedMessage>,
  b: MessageEvent<StampedMessage>,
): number {
  const timeA = getTimestampForMessage(a.message);
  const timeB = getTimestampForMessage(b.message);
  if (timeA == undefined || timeB == undefined) {
    if (timeA == undefined && timeB == undefined) {
      return 0;
    } else if (timeA == undefined) {
      return -1;
    } else {
      return 1;
    }
  }
  return compare(timeA, timeB);
}
```

Reply:
Good optimization. Thanks!